Sources/Stream-Feed-UIKit-IOS/Views/Collection View Cells/VideoCell.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  VideoCell.swift
+//  
+//
+//  Created by Sherif Shokry on 16/10/2023.
+//
+
+import UIKit
+import AVKit
+import Reusable
+
+public final class VideoCell: UICollectionViewCell, NibReusable {
+    
+    @IBOutlet weak var showControlsButton: UIButton!
+    @IBOutlet weak var playButtonView: UIView!
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    @IBOutlet weak var videoView: UIView!
+    @IBOutlet public weak var removeButton: UIButton!
+    var streamingURL: String?
+    var logErrorAction: ((String, String) -> Void)?
+    private(set) var videoPlayerWrapper: VideoPlayerWrapperProtocol = VideoPlayerWrapper()
+    var showControlsState: ((Bool) -> Void)?
+    var videoPlayerTapped: (() -> Void)?
+    
+    public override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+        
+        videoPlayerWrapper.logErrorAction = logErrorAction
+        activityIndicator.removeFromSuperview()
+        activityIndicator.stopAnimating()
+    }
+    
+    public override func prepareForReuse() {
+        super.prepareForReuse()
+        self.videoPlayerWrapper.playerController?.showsPlaybackControls = false
+        self.showControlsState?(self.videoPlayerWrapper.playerController?.showsPlaybackControls ?? false)
+        self.videoPlayerWrapper.player?.pause()
+        self.videoPlayerWrapper.playerController?.player?.seek(to: .zero) { [weak self] _ in
```

Comment:
@SherifShokry123 I feel like the seek to .zero then pause isn't necessary here
and this might cause some hanging as it was causing the same on didScroll func in our client side codebase

Reply:
I suggest to seek to zero on click of the video before playing it and just pause the video on reuse